### ISSUE TITLE
Values of identifierUris property must use a verified domain of the organization or its subdomain

### ISSUE THREAD

**Issue by rogermscarvalho** [LockTar/AzureAdApplicationManagement#63]
"Set Azure AD Application" task fails with error: "Values of identifierUris property must use a verified domain of the organization or its subdomain: 'https://APP_REGISTRATION_NAME'". It seems the task is passing the value of field "Name" instead of field "App Id Uri", once I generate the Identify URI in a previous task and it has the format "https://inoffice.onmicrosoft.com/$([guid]::newguid())/user_impersonation". It was working properly till last week.

Logs:
2021-10-20T17:33:15.4464307Z ##[debug]Exception:
2021-10-20T17:33:15.4498688Z ##[debug]System.Exception: Values of identifierUris property must use a verified domain of the organization or its subdomain: 'https://AZU-EUR-WW-DEV-CMS930-FE-SOLRAH'
2021-10-20T17:33:15.4545102Z ##[error]Values of identifierUris property must use a verified domain of the organization or its subdomain: 'https://AZU-EUR-WW-DEV-CMS930-FE-SOLRAH'
2021-10-20T17:33:15.4547229Z ##[debug]Processed: ##vso[task.logissue type=error]Values of identifierUris property must use a verified domain of the organization or its subdomain: 'https://AZU-EUR-WW-DEV-CMS930-FE-SOLRAH'
2021-10-20T17:33:15.4560402Z ##[debug]Processed: ##vso[task.complete result=Failed]

**Comment by LockTar** [comment]
Hi @rogermscarvalho, thank you for reporting this issue. This is not really an issue but more a new feature of the AD. See for more information the [release notes](https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-breaking-changes#appid-uri-in-single-tenant-applications-will-require-use-of-default-scheme-or-verified-domains). 

The task is doing an update so it will encounter the new validation rule that went active the 15th of this month. 

I think I have to implement a breaking change. But I need to think this through when I have more time. 

Any suggestions?

**Comment by wouterroos** [comment]
Hi,

I also ran into this issue using the V2 step today. However, when I upgraded to V3 I'm still getting the following error:

`##[error]Values of identifierUris property must use a verified domain of the organization or its subdomain: 'https://localhost'`

The error is slightly different from when I used V2, which was:

`2021-11-15T09:21:52.7140798Z ##[error]Values of identifierUris property must use a verified domain of the organization or its subdomain: 'https://***/[APP_REGISTRATION_NAME]'` 

Am I missing something?

Thanks!

**Comment by LockTar** [comment]
@wouterroos Between v2 and v3 of the task the underlying libraries are changed. So that can be the reason why you are getting slightly different error messages. 
The cause of this error message is still the same. You need a verified domain. See the release notes of the Azure AD. This is a change from Microsoft and not from this task.
See the [above comment](https://github.com/LockTar/AzureAdApplicationManagement/issues/63#issuecomment-947936056) from this issue.

**Comment by wouterroos** [comment]
@LockTar Thanks, I changed the reply URL to the default .onmicrosoft.com domain of our tenant, and everything is now working as before.

**Comment by LockTar** [comment]
@wouterroos so you mean in example: `https://{tenant name}.onmicrosoft.com/{app registration name}`.
If so, this can help other people with this problem 👍🏻

**Comment by wouterroos** [comment]
@LockTar Yes exactly, any domain registered on the AD tenant should work but since this one is there by default it could indeed be an easy fix for people running into this issue!
